lib/cambiatus/kyc/kyc_data.ex

```diff
@@ -0,0 +1,125 @@
+defmodule Cambiatus.Kyc.KycData do
+  @moduledoc """
+  KYC Ecto Model
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Cambiatus.{Accounts.User, Kyc.Country, Repo}
+
+  schema "kyc_data" do
+    field(:user_type, :string)
+    field(:document, :string)
+    field(:document_type)
+    field(:phone, :string)
+    field(:is_verified, :boolean)
+
+    belongs_to(:account, User, references: :account, type: :string)
+    belongs_to(:country, Country)
+
+    timestamps()
+  end
+
+  @required_fields ~w(account_id user_type document document_type phone country_id)a
+  @optional_fields ~w(is_verified)
+
+  def changeset(model, params \\ :empty) do
+    model
+    |> Repo.preload(:country)
+    |> Repo.preload(:account)
+    |> cast(params, @required_fields, @optional_fields)
+    |> validate_required(@required_fields)
+    |> validate_user_type()
+    |> validate_document_type()
+    |> validate_format(:phone, ~r/[1-9]{1}\d{3}-?\d{4}/)
+    |> validate_document()
+  end
+
+  def validate_user_type(changeset) do
+    if Enum.any?(["natural", "juridical"], &(&1 == get_field(changeset, :user_type))) do
+      changeset
+    else
+      add_error(changeset, :user_type, "is invalid")
+    end
+  end
+
+  def validate_document_type(changeset) do
+    if Enum.any?(
+         ["mipyme", "gran_empresa", "cedula_de_identidad", "dimex", "nite"],
+         &(&1 == get_field(changeset, :document_type))
+       ) do
+      changeset
+    else
+      add_error(changeset, :document_type, "is invalid")
+    end
+  end
+
+  @doc """
+  This function validates documents that follow the Costa Rica's standard
+
+  Reference: https://help.hulipractice.com/es/articles/1348413-ingresar-informacion-de-emisores-solo-para-costa-rica
+  """
+  def validate_document(%{valid?: false} = changeset), do: changeset
+
+  def validate_document(changeset) do
+    user_type = get_field(changeset, :user_type)
+
+    changeset
+    |> validate_document_by_user_type(user_type)
+    |> validate_document_by_document_type()
+  end
+
+  defp validate_document_by_user_type(changeset, "natural") do
+    document_type = get_field(changeset, :document_type)
+    natural_documents = ["cedula_de_identidad", "dimex", "nite"]
+
+    if Enum.any?(natural_documents, &(&1 == document_type)) do
```

Comment:
We could use `Enum.member` here instead of `Enum.any`. It prevents an allocation and is likely faster overall, since there's no overhead of calling a function for every member.

Reply:
```suggestion
    if document_type in natural_documents do
```
This apparently works as well :)

Reply:
awesome, didn't knew about this operator! 🤯